src/mbgl/layout/symbol_instance.cpp

```diff
@@ -88,12 +88,11 @@ SymbolInstance::SymbolInstance(Anchor& anchor_,
                                const float iconRotation,
                                const float textRotation,
                                float radialTextOffset_,
-                               bool allowVerticalPlacement) :
+                               bool allowVerticalPlacement,
+                               const SymbolContent& iconType) :
     sharedData(std::move(sharedData_)),
     anchor(anchor_),
-    // 'hasText' depends on finding at least one glyph in the shaping that's also in the GlyphPositionMap
-    hasText(!sharedData->empty()),
-    hasIcon(shapedIcon),
+    symbolContent(static_cast<uint8_t>(iconType)),
```

Comment:
instead of `static_cast` all the time we could introduce:
```

MBGL_CONSTEXPR SymbolContent operator|(SymbolContent a, SymbolContent b) {
    return SymbolContent(mbgl::underlying_type(a) | mbgl::underlying_type(b));
}

MBGL_CONSTEXPR SymbolContent& operator|=(SymbolContent& a, SymbolContent b) {
    return (a = a | b);
}

MBGL_CONSTEXPR SymbolContent operator&(SymbolContent a, SymbolContent b) {
    return SymbolContent(mbgl::underlying_type(a) & mbgl::underlying_type(b));
}
```